doc/api/esm.md

```diff
@@ -106,11 +106,21 @@ fs.readFile('./foo.txt', (err, body) => {
 <!-- type=misc -->
 
 To customize the default module resolution, loader hooks can optionally be
-provided via a `--loader ./loader-name.mjs` argument to Node.
+provided via a `--loader ./loader-name.mjs` argument to Node. This argument
+can be passed multiple times to compose loaders like
+`--loader ./loader-coverage.mjs --loader ./loader-mocking.mjs`. The last loader
+will be used for module loading and must explicitly call to the parent loader
+in order to provide compose behavior.
 
 When hooks are used they only apply to ES module loading and not to any
 CommonJS modules loaded.
 
+All loaders are created by invoking the default export of their module as a
+function. The parameters given to the the function is a single object with
+properties to call the `resolve` and `dynamicInstantiate` hooks of the parent
+loader. The default loader has a `resolve` hook and `null` for the value
+`dynamicInstantiate`.
+
```

Comment:
👆 By default loader do you mean the last loader (far right)? Or something internal?

Reply:
internal.

Reply:
he's referring to the default internal loader that is used if you don't hook anything

Reply:
Should the default dynamic instantiate be a function that just throws "no instantiate hook"? Then each caller can assume chaining, with the fallthrough being an error naturally.

Reply:
Since it doesn't impact the user exposed code and not something the user can observe (is that right?) would that make more sense as a side note.

Reply:
to the user its just a `parent` with `{ resolve: [Function], dynamicInstantiate: null }`

Reply:
Specifically, the user should be able to chain the first loader identically to any other loader, so they should specifically call the `dynamicInstantiate` if creating their own to ensure chaining. Thus the first one as throwing is needed to maintain this equivalence (although the loader needn't necessarily need to know that it is throwing).

Reply:
I've change the `dynamicInstantiate` to be a throwing function.